channeldb/graph.go

```diff
@@ -1279,6 +1350,122 @@ func (c *ChannelGraph) HasChannelEdge(
 	return upd1Time, upd2Time, exists, isZombie, nil
 }
 
+func (c *ChannelGraph) HasChannelEdge2(
+	chanID uint64) (uint32, uint32, bool, bool, error) {
```

Comment:
I think time for a return value struct given we have 5 return values here.

Reply:
Also these two methods look pretty much identical (just height vs time stamps). Minimally, we can refactor some of the duplicate inner logic into a new function. Going a step further: can we abstract away the height vs timestamp difference? So something like a `UpdateOrder`?